### PR TITLE
Document Conduit's Automatic TLS feature

### PR DESCRIPTION
**Pull request by olix0r**

### CODE REVIEW

File: doc/automatic-tls.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+The conduit control plane must be installed with the `--tls=optional` flag:
+
+```
+conduit install --tls=optional |kubectl apply -f -
+```
+
+This causes a Certificate Authority (CA) container to be run in the control
+plane. The CA is responsible for watching the Kubernetes API for new
+Conduit-enabled pods and, as new pods are created, it ensures that the pod
```
Comment (briansmith) [inline doc/automatic-tls.md]: I suggest replacing "The CA is responsible..." with:

The CA watches for the creation and updates of Conduit-enabled pods. For each Conduit-enabled pod, it generates a private key, issues a certificate, and distributes the certificate and private key to each pod as a Kubernetes secret.

File: doc/automatic-tls.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+title = "Experimental: Automatic TLS"
+docpage = true
+[menu.docs]
+    parent = "automatic-tls"
++++
+
+As of [Conduit v0.5.0][conduit-v0.5.0], Conduit can be configured to automatically
+instrument applications to communicate with Transport Layer Security (TLS).
+
+When TLS enabled, Conduit automatically establishes and authenticates
```
Comment (klingerf) [inline doc/automatic-tls.md]: Typo here: `When TLS enabled` => `When TLS is enabled`

File: doc/automatic-tls.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+secure,private connections between Conduit proxies. This is done without
```
Comment (klingerf) [inline doc/automatic-tls.md]: Missing space: `secure,private` => `secure, private`

File: doc/automatic-tls.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+breaking unencrypted communication with endpoints that are not configured
+with TLS-enabled Conduit proxies.
+
+This feature is currently **experimental** and is designed to _fail open_ so
+that it cannot easily break existing applications. As the feature matures,
+this policy will change in favor of stronger security guarantees.
+
+### Getting started with TLS
+
+
```
Comment (klingerf) [inline doc/automatic-tls.md]: I'd remove this extra blank line

File: doc/automatic-tls.md (hunk 1 above)
Comment (klingerf) [inline doc/automatic-tls.md]: I think this needs one more comma:

```
Conduit-enabled pods, and, as new pods...
```

File: doc/automatic-tls.md [hunk 5]
```diff
@@ -0,0 +1,62 @@
++++
+title = "Experimental: Automatic TLS"
+docpage = true
+[menu.docs]
+    parent = "automatic-tls"
++++
+
+As of [Conduit v0.5.0][conduit-v0.5.0], Conduit can be configured to automatically
+instrument applications to communicate with Transport Layer Security (TLS).
```
Comment (wmorgan) [inline doc/automatic-tls.md]: "instrument" doesn't seem quite right. Maybe "allow" or "enable"

File: doc/automatic-tls.md [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+breaking unencrypted communication with endpoints that are not configured
+with TLS-enabled Conduit proxies.
+
+This feature is currently **experimental** and is designed to _fail open_ so
+that it cannot easily break existing applications. As the feature matures,
+this policy will change in favor of stronger security guarantees.
+
+### Getting started with TLS
+
+The conduit control plane must be installed with the `--tls=optional` flag:
```
Comment (wmorgan) [inline doc/automatic-tls.md]: I think it's worth a sentence or paragraph somewhere that explains what `optional` signifies (possibly by contrasting against other, future modes of operation). If I came into this blind, I would expect it to be called "enabled" or "on".

Comment (briansmith) [reply doc/automatic-tls.md]: "optional" is used to indicate the "fail-open" behavior.

**Pull request merged**